pkg/upgrade/control_plane.go

```diff
@@ -1182,7 +1182,11 @@ func (u *ControlPlaneUpgrader) etcdctl(ctx context.Context, args ...string) (str
 func (u *ControlPlaneUpgrader) etcdctlForPod(ctx context.Context, pod *v1.Pod, args ...string) (string, string, error) {
 	u.log.Info("Running etcdctl", "pod", pod.Name, "args", strings.Join(args, " "))
 
-	endpoint := fmt.Sprintf("https://%s:2379", pod.Status.PodIP)
+	ip := pod.Status.PodIP
+	if ip == "" {
+		ip = "[127.0.0.1]"
```

Comment:
Is this the format the endpoint is expecting? With the square brackets around it?

Reply:
It actually works both ways. I wound up changing it to this based on the `etcdctl` command found in the liveness probe for the pod:

```
      exec:
        command:
        - /bin/sh
        - -ec
        - ETCDCTL_API=3 etcdctl --endpoints=https://[127.0.0.1]:2379 --cacert=/etc/kubernetes/pki/etcd/ca.crt
          --cert=/etc/kubernetes/pki/etcd/healthcheck-client.crt --key=/etc/kubernetes/pki/etcd/healthcheck-client.key
          get foo
```

Reply:
Got it, does `pod.Status.PodIP` also come in brackets? If not, I'd prefer to keep it consistent

Reply:
Fair point - it won't, no. I'll go ahead and tweak that.